manifests/params.pp

```diff
@@ -2,10 +2,14 @@
 # @api private
 class dhcp::params {
 
-  $dnsdomain = [$facts['domain']]
+  $dnsdomain = $facts['os']['family'] ? {
+    'Archlinux' => [$facts['domain']], # lint:ignore:legacy_facts
```

Comment:
This usually means the wrong facts are loaded in rspec-puppet-facts. Yet another reason to separate changes into smaller PRs. Combining it means we'd need to fix this, which holds up EL8 support.